File: samples/Sample.py
"""Islands Dark Theme - Python Sample"""

import sys
from typing import List, Optional
from dataclasses import dataclass, field

MAX_VALUE = 1000
VERSION = "1.0.0"


@dataclass
class DataPoint:
    """Represents a single data point."""
    x: float
    y: float
    label: str = ""
    metadata: dict = field(default_factory=dict)

    def distance(self, other: "DataPoint") -> float:
        return ((self.x - other.x) ** 2 + (self.y - other.y) ** 2) ** 0.5


def process_items(items: List[str], prefix: str = "item") -> List[str]:
    """Process a list of items with optional prefix."""
    result = []
    for i, item in enumerate(items):
        result.append(f"{prefix}_{i}: {item}")
    return result


async def fetch_data(url: str) -> Optional[dict]:
    """Fetch data from a URL asynchronously."""
    import aiohttp
    async with aiohttp.ClientSession() as session:
        async with session.get(url) as response:
            if response.status == 200:
                return await response.json()
    return None


class ColorTheme:
    BG = "#191A1C"
    FG = "#BCBEC4"
    KEYWORD = "#CF8E6D"

    def __init__(self, name: str):
        self.name = name
        self._colors: dict = {}

    @property
    def is_dark(self) -> bool:
        return True

    @staticmethod
    def from_config(path: str) -> "ColorTheme":
        with open(path, "r") as f:
            lines = f.readlines()
        return ColorTheme(lines[0].strip())


if __name__ == "__main__":
    theme = ColorTheme("Islands Dark")
    items = process_items(["alpha", "beta", "gamma"])
    # This is a line comment
    print(f"Theme: {theme.name}, Items: {len(items)}")

File: samples/Sample.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace IslandsDark.Sample
{
    public class ColorTheme
    {
        public const string Version = "1.0.0";
        public const int MaxColors = 256;

        private readonly Dictionary<string, string> _colors = new()
        {
            ["background"] = "#191A1C",
            ["foreground"] = "#BCBEC4",
            ["keyword"] = "#CF8E6D",
            ["string"] = "#6AAB73",
            ["number"] = "#2AACB8"
        };

        public string Name { get; init; }
        public bool IsDark => true;

        public ColorTheme(string name)
        {
            Name = name;
        }

        public string this[string key]
        {
            get => _colors.TryGetValue(key, out var value) ? value : null;
            set => _colors[key] = value;
        }

        /// <summary>
        /// Applies the theme to the specified target.
        /// </summary>
        public async Task ApplyAsync(string target)
        {
            await Task.Delay(100);
            Console.WriteLine($"Applied {Name} to {target}");
        }

        public static ColorTheme Default => new("Islands Dark");
    }

    public record Point(double X, double Y)
    {
        public double DistanceTo(Point other) =>
            Math.Sqrt(Math.Pow(X - other.X, 2) + Math.Pow(Y - other.Y, 2));
    }

    #region Extensions
    public static class ColorExtensions
    {
        public static string ToHex(this (int r, int g, int b) color) =>
            $"#{color.r:X2}{color.g:X2}{color.b:X2}";
    }
    #endregion
}